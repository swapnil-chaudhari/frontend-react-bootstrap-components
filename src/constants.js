export const FILTER_PERIODS = {
    YEAR: 365,
    MONTH: 27,
    WEEK: 6,
    DAYS_90: 90,
};

export const SEARCH_TYPES = {
    ARTISTS: 'artists',
    TRACKS: 'tracks'
};

export const KEYS = {
    ENTER: 13,
};
