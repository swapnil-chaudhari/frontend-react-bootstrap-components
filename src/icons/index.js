import './icons.scss';

export { default as Aan } from './aan';
export { default as Advertise } from './advertise';
export { default as AppleMusic } from './apple-music';
export { default as ArtworkEmpty } from './artwork-empty';
export { default as Attention } from './attention';
export { default as AudioDisabled } from './audio-disabled';
export { default as AudioEnabled } from './audio-enabled';
export { default as AudioPlayerMute } from './audio-player-mute';
export { default as AudioPlayerNext } from './audio-player-next';
export { default as AudioPlayerNextLite } from './audio-player-next-lite';
export { default as AudioPlayerPause } from './audio-player-pause';
export { default as AudioPlayerPauseLite } from './audio-player-pause-lite';
export { default as AudioPlayerPlay } from './audio-player-play';
export { default as AudioPlayerPrevious } from './audio-player-previous';
export { default as AudioPlayerPreviousLite } from './audio-player-previous-lite';
export { default as AudioPlayerVolume } from './audio-player-volume';
export { default as Calendar } from './calendar';
export { default as Camera } from './camera';
export { default as Chevron } from './chevron';
export { default as CleanVersion } from './clean-version';
export { default as Close } from './close';
export { default as CloseRound } from './close-round';
export { default as CompleteCheckmark } from './complete-checkmark';
export { default as Digital } from './digital';
export { default as DigitalAudio } from './digital-audio';
export { default as DownDirectionArrow } from './down-direction-arrow';
export { default as DownloadBox } from './download-box';
export { default as DownloadSymbol } from './download-symbol';
export { default as Edit } from './edit';
export { default as Error404 } from './error-404.js';
export { default as Explicit } from './explicit';
export { default as ExternalLink } from './external-link';
export { default as Facebook } from './facebook';
export { default as FilledChevronDown } from './filled-chevron-down';
export { default as FilledChevronUp } from './filled-chevron-up';
export { default as Film } from './film';
export { default as Gear } from './gear.js';
export { default as GoToCatalog } from './go-to-catalog';
export { default as History } from './history';
export { default as IncompleteAlert } from './incomplete-alert';
export { default as ImportArrow } from './import-arrow';
export { default as Info } from './info';
export { default as Instagram } from './instagram';
export { default as LeftPaginationArrow } from './left-pagination-arrow';
export { default as LoadingArrows } from './loading-arrows';
export { default as LoadingRing } from './loading-ring';
export { default as Logo } from './logo';
export { default as LogoAndName } from './logo-and-name.js';
export { default as Maintenance } from './maintenance.js';
export { default as Man } from './man';
export { default as NewProduct } from './new-product';
export { default as Pdf } from './pdf';
export { default as Physical } from './physical';
export { default as PhysicalAudio } from './physical-audio';
export { default as Play } from './play';
export { default as PlaylistPlaceholder } from './playlist-placeholder';
export { default as Plus } from './plus';
export { default as Refresh } from './refresh';
export { default as RightDirectionArrow } from './right-direction-arrow';
export { default as RightPaginationArrow } from './right-pagination-arrow';
export { default as RoyaltyShareLogo } from './royalty-share-logo';
export { default as Search } from './search';
export { default as ServerError } from './server-error';
export { default as Spotify } from './spotify';
export { default as StatusComplete } from './status-complete';
export { default as StatusInProgress } from './status-in-progress';
export { default as StatusRed } from './status-red';
export { default as StatusSubmitted } from './status-submitted';
export { default as Stop } from './stop';
export { default as SuccessCheckmark } from './success-checkmark';
export { default as ThreeDots } from './three-dots';
export { default as ThreeRightDirectedArrows } from './three-right-directed-arrows';
export { default as TracksCompleteCheckmark } from './tracks-complete-checkmark';
export { default as TracksEmpty } from './tracks-empty';
export { default as TransparentLogo } from './transparent-logo';
export { default as TrashCan } from './trash-can';
export { default as TripleCheckmark } from './triple-checkmark';
export { default as Twitter } from './twitter';
export { default as UpAndDownArrows } from './up-and-down-arrows';
export { default as UpAndDownCarets } from './up-and-down-carets';
export { default as UpDirectionArrow } from './up-direction-arrow';
export { default as Upload } from './upload';
export { default as UploadArrow } from './upload-arrow';
export { default as VideoError } from './video-error';
export { default as Video } from './video';
export { default as ViewMoreNextArrow } from './view-more-next-arrow';
export { default as Woman } from './woman';
export { default as Xls } from './xls';
export { default as YouTube } from './youtube';
