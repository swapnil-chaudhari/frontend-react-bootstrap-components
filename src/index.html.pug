html
    head
        meta(charset="utf-8")
        meta(http-equiv="X-UA-Compatible", content="IE=edge,chrome=1")
        title="Shared Components Library Playground"

        link(rel='stylesheet', href=__SHARED_ASSET_URL__ + '/css/vendor-redesign.css')
        link(rel='stylesheet', href=__SHARED_ASSET_URL__ + '/css/global-redesign.css')
        link(rel='stylesheet', href=__DEPLOYED_URL__ + 'react-components.css')

        link(rel='stylesheet', href='//cdnjs.cloudflare.com/ajax/libs/codemirror/5.0.0/codemirror.min.css')
        link(rel='stylesheet', href='//cdnjs.cloudflare.com/ajax/libs/codemirror/5.0.0/theme/monokai.min.css')
    body
        div(id='playground')
        script(type='text/javascript', src='//cdnjs.cloudflare.com/ajax/libs/codemirror/5.0.0/codemirror.min.js')
        script(type='text/javascript', src='//cdnjs.cloudflare.com/ajax/libs/codemirror/5.0.0/mode/javascript/javascript.min.js')
        script(src='/frontend-react-components/main.bundle.js')
