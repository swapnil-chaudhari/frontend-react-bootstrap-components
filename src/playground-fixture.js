/* global React, ReactDOM, Button, mountNode */
const Playground = () => (
    <Button bsStyle="success">Click me</Button>
);

ReactDOM.render(<Playground />, mountNode);
