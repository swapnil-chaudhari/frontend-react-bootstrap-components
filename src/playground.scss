.Playground {
    margin: 25px;

    .playgroundPreview { /* stylelint-disable-line */
        margin-top: 25px;
    }
}
