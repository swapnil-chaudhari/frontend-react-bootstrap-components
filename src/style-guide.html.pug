html
    head
        meta(charset="utf-8")
        meta(http-equiv="X-UA-Compatible", content="IE=edge,chrome=1")
        title="Component Style Guide"

        link(rel='stylesheet', href=__SHARED_ASSET_URL__ + '/css/vendor-redesign.css')
        link(rel='stylesheet', href=__SHARED_ASSET_URL__ + '/css/global-redesign.css')
        link(rel='stylesheet', href=__DEPLOYED_URL__ + 'react-components.css')
    body
        div(id='style-guide')
        script(src=__DEPLOYED_URL__ + 'style-guide.bundle.js')
