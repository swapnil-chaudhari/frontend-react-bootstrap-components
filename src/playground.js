import React from 'react';
import ReactDOM, { render } from 'react-dom';
import PlaygroundCode from 'component-playground';
import * as Components from 'src/index';
import './playground.scss';

const initialText = require('raw-loader!./playground-fixture');

const Playground = () => (
    <div className="Playground">
        <PlaygroundCode
            codeText={ initialText }
            scope={ { React, ReactDOM, ...Components } }
            noRender={ false }
        />
        <Components.SVGDefinitions />
    </div>
);

render(<Playground />, document.getElementById('playground'));
