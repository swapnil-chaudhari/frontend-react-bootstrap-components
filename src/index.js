import * as Icons from './icons';
import {
    Tooltip,
    Overlay,
    OverlayTrigger,
    Nav,
    NavItem,
    MenuItem,
    NavDropdown,
    Popover
} from 'react-bootstrap';

// Analytics components
export { default as Snapshot } from './snapshot/snapshot';
export { default as SnapshotItem } from './snapshot/snapshot-item';

// Bootstrap and bootstrap-based components
export { default as AlertMessage } from './alert-message/alert-message';
export { default as Button } from './button/button';
export { default as LoadingButton } from './loading-button/loading-button';
export { default as ButtonsRow } from './buttons-row/buttons-row';
export { default as CollapsableAlert } from './collapsable-alert/collapsable-alert';
export { default as DismissibleAlert } from './dismissible-alert/dismissible-alert';
export { default as DropdownButton } from './dropdown-button/dropdown-button';
export {
    MenuItem,
    Nav,
    NavDropdown,
    NavItem,
    Overlay,
    OverlayTrigger,
    Popover,
    Tooltip
};

// Chart components
export { default as ChartContainer } from './chart/chart-container';
export { default as ChartRectClipPath } from './chart/chart-rect-clip-path';
export { default as ChartOverlayTrigger } from './chart/chart-overlay-trigger';
export { default as ChartXAxis } from './chart/chart-x-axis.js';
export { default as ChartYAxis } from './chart/chart-y-axis';
export { default as ChartGrid } from './chart/chart-grid';
export { default as ChartLine } from './chart/chart-line';
export { default as ChartGraph } from './chart/chart-graph';
export { default as ChartGraphContainer } from './chart/chart-graph-container';

export { default as DoughnutChart } from './doughnut-chart/doughnut-chart';
export { default as GroupedBarChart } from './grouped-bar-chart/grouped-bar-chart';

// Form field components
export {
    default as CharacterLimitIndicator
} from './character-limit-indicator/character-limit-indicator';
export { default as Checkbox } from './checkbox/checkbox';
export { default as DayInput } from './day-input/day-input';
export { default as SingleDateInput } from './single-date-input/single-date-input';
export { default as RangeDateInput } from './range-date-input/range-date-input';
export { default as FormField } from './form-field/form-field';
export { default as RadioButton } from './radio-button-group/radio-button/radio-button';
export { default as RadioButtonGroup } from './radio-button-group/radio-button-group';
export { default as SelectInput } from './select-input/select-input';
export {
    default as SelectInputWithAddNew
} from './select-input-with-add-new/select-input-with-add-new';
export { default as TagsInput } from './tags-input/tags-input';
export { default as FullWidthButton } from './full-width-button/full-width-button';

// Iconography
export { default as Help } from './help/help';
export { default as Icon } from './icon/icon';
export { Icons };
export { default as SVGDefinitions } from './svg-definitions/svg-definitions';

// Loading states
export { default as LoadingIndicator } from './loading-indicator/loading-indicator';
export { default as LoadingStateWrapper } from './loading-state-wrapper/loading-state-wrapper';

// Navigation items
export { default as MainNavigation } from './main-navigation/main-navigation';
export { default as Pagination } from './pagination/pagination';
export { default as PaginationItem } from './pagination/pagination-item';
export { default as SubNavigation } from './sub-navigation/sub-navigation';
export { default as SubNavigationItem } from './sub-navigation/sub-navigation-item';

// Track/Artist search
export { default as SearchResults } from './search-row/search-results/search-results';
export { default as SearchRow } from './search-row/search-row';

// Tables
export { default as ClickableTable } from './clickable-table/clickable-table';

// Tile
export { default as Tile } from './tile/tile';

// Territories
export {
    default as TerritoriesListManager
} from './territories-list-manager/territories-list-manager';
export {
    default as ListHeaderSearch
} from './territories-list-manager/list-header-search/list-header-search';

export { default as MaintenanceScreen } from './maintenance-screen/maintenance-screen.js';
export { default as Modal } from './modal/modal.js';
export { default as ToggleInput } from './toggle-input/toggle-input.js';
