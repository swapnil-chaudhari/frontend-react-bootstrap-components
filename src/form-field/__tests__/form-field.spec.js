import FormField from '../form-field';
import { Help } from 'src/index';
import React from 'react';
import renderShallow from 'render-shallow';
import { findWithType } from 'react-shallow-testutils';

describe('<FormField>', () => {
    describe('when it renders with required properties', () => {

        describe('and its child is an html input', () => {
            let component;

            beforeAll(() => {
                component = renderShallow(
                    <FormField>
                        <input />
                    </FormField>
                ).output;
            });

            test('renders a form field wrapping the supplied child component', () => {
                expect(component.type).toEqual('div');
                expect(component.props.className).toEqual('form-group');
                expect(findWithType(component, 'label').props.htmlFor).toMatch(/^\d+$/);
            });
        });

        describe('and its child is a React Component', () => {
            const ChildComponent = () => <input />;
            let component;

            beforeAll(() => {
                component = renderShallow(
                    <FormField>
                        <ChildComponent />
                    </FormField>
                ).output;
            });

            test('renders a form field wrapping the supplied child component', () => {
                expect(component.type).toEqual('div');
                expect(component.props.className).toEqual('form-group');
                expect(findWithType(component, 'label').props.htmlFor)
                    .toMatch(/^\d+$/);
            });
        });

    });

    describe('when it is rendered with a single child array', () => {
        let component;

        beforeAll(() => {
            component = renderShallow(
                <FormField>
                    { [<input key="1" id="its-ian" />] }
                </FormField>
            ).output;
        });

        test('renders the child', () => {
            expect(component).toBeTruthy();
        });

    });

    describe('when it is rendered with multiple children', () => {
        let render;

        beforeAll(() => {
            render = () => renderShallow(
                <FormField>
                [
                    <input type="text" key="1" />,
                    <input type="text" key="2" />
                ]
                </FormField>
            );
        });

        test('throws an error', () => {
            expect(render).toThrow();
        });

    });

    describe('when it has a value for props.className', () => {
        let component;

        beforeAll(() => {
            component = renderShallow(
                <FormField className="test-class">
                    <div id="test-div"></div>
                </FormField>
            ).output;
        });

        test(
            'augments the style classes on the outermost div with the supplied classes',
            () => {
                expect(component).toMatchSnapshot();
            }
        );
    });

    describe('when the child component has a value for props.className', () => {
        const childClass = 'child-class';
        let component;

        beforeAll(() => {
            component = renderShallow(
                <FormField>
                    <input className={ childClass } />
                </FormField>
            ).output;
        });

        test(
            "will augment the child's props.className with the FormField-form-control class",
            () => {
                expect(component).toMatchSnapshot();
            }
        );
    });

    describe('when the child component has a value for props.id', () => {
        const childId = 'child-id';
        let component;

        beforeAll(() => {
            component = renderShallow(
                <FormField>
                    <input id={ childId } />
                </FormField>
            ).output;
        });

        test('will preserve the child component\'s id', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when props.label is supplied', () => {
        const props = {
            label: 'label'
        };
        let component;

        beforeAll(() => {
            component = renderShallow(
                <FormField { ...props }>
                    <input />
                </FormField>
            ).output;
        });

        test('renders a labeled input', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when props.help is supplied', () => {
        describe('and it is simple text', () => {
            const props = {
                help: 'Please help me, please.'
            };
            let component;

            beforeAll(() => {
                component = renderShallow(
                    <FormField { ...props }>
                        <input />
                    </FormField>
                ).output;
            });

            test('renders a help icon', () => {
                expect(component).toMatchSnapshot();
            });
        });

        describe('and it is React.element', () => {
            const props = {
                help: <Help id="help" message="Please help me, please." />
            };

            let component;

            beforeAll(() => {
                component = renderShallow(
                    <FormField { ...props }>
                        <input />
                    </FormField>
                ).output;
            });

            test('renders a help icon', () => {
                expect(component).toMatchSnapshot();
            });
        });
    });

    describe('when props.isRequired is passed in', () => {

        describe('and it is true', () => {
            const props = {
                isRequired: true
            };
            let component;

            beforeAll(() => {
                component = renderShallow(
                    <FormField { ...props }>
                        <input />
                    </FormField>
                ).output;
            });

            test('renders a required icon and title', () => {
                expect(component).toMatchSnapshot();
            });

        });

        describe('and it is false', () => {
            const props = {
                isRequired: false
            };
            let component;

            beforeAll(() => {
                component = renderShallow(
                    <FormField { ...props }>
                        <input />
                    </FormField>
                ).output;
            });

            test('does not render a required icon and title', () => {
                expect(component).toMatchSnapshot();
            });

        });

    });


    describe('when props.help and props.isRequired are supplied', () => {
        const props = {
            help: 'Please help me, please.',
            isRequired: true
        };
        let component;

        beforeAll(() => {
            component = renderShallow(
                <FormField { ...props }>
                    <input />
                </FormField>
            ).output;
        });

        test('renders help and required icon in the proper order', () => {
            expect(component).toMatchSnapshot();
        });
    });


    describe('when props note is', () => {
        describe('supplied and it', () => {
            const props = {
                note: 'When I wore a younger mans clothes'
            };
            let component;

            beforeAll(() => {
                component = renderShallow(
                    <FormField { ...props }>
                        <input />
                    </FormField>
                ).output;
            });

            test('renders a note', () => {
                expect(component).toMatchSnapshot();
            });

        });
    });

    describe('when props errorMessage is', () => {
        describe('supplied and it simple string', () => {
            const props = {
                errorMessage: 'Everybody knows ducks come from volcanoes.'
            };
            let component;

            beforeAll(() => {
                component = renderShallow(
                    <FormField { ...props }>
                        <input />
                    </FormField>
                ).output;
            });

            test('renders an error message', () => {
                expect(component).toMatchSnapshot();
            });

        });

        describe('supplied and it is node', () => {
            const props = {
                errorMessage: (
                    <span>Everybody knows ducks come from volcanoes.</span>
                )
            };
            let component;

            beforeAll(() => {
                component = renderShallow(
                    <FormField { ...props }>
                        <input />
                    </FormField>
                ).output;
            });

            test('renders an error message', () => {
                expect(component).toMatchSnapshot();
            });

        });
    });

    describe('when props.label and labelAppendix are supplied', () => {
        const props = {
            label: 'label',
            labelAppendix: <a>appendix</a>
        };
        let component;

        beforeAll(() => {
            component = renderShallow(
                <FormField { ...props }>
                    <input />
                </FormField>
            ).output;
        });

        test('renders a label with an appendix', () => {
            expect(component).toMatchSnapshot();
        });
    });


});
