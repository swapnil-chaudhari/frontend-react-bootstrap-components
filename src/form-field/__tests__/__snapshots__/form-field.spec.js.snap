// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormField> when it has a value for props.className augments the style classes on the outermost div with the supplied classes 1`] = `
<div
  className="form-group test-class"
  id={undefined}
>
  <label
    className="FormField-label"
    htmlFor="test-div"
  />
  <div
    className="FormField-form-control"
    id="test-div"
  />
  <span
    className="FormField-note"
  />
</div>
`;

exports[`<FormField> when props errorMessage is supplied and it is node renders an error message 1`] = `
<div
  className="form-group FormField-group-has-error"
  id={undefined}
>
  <label
    className="FormField-label"
    htmlFor="737884497096766615"
  />
  <input
    className="FormField-form-control"
    id="737884497096766615"
  />
  <span
    className="FormField-note"
  />
  <span
    className="FormField-errorMessage"
  >
    <span>
      Everybody knows ducks come from volcanoes.
    </span>
  </span>
</div>
`;

exports[`<FormField> when props errorMessage is supplied and it simple string renders an error message 1`] = `
<div
  className="form-group FormField-group-has-error"
  id={undefined}
>
  <label
    className="FormField-label"
    htmlFor="737884497096766615"
  />
  <input
    className="FormField-form-control"
    id="737884497096766615"
  />
  <span
    className="FormField-note"
  />
  <span
    className="FormField-errorMessage"
  >
    Everybody knows ducks come from volcanoes.
  </span>
</div>
`;

exports[`<FormField> when props note is supplied and it renders a note 1`] = `
<div
  className="form-group"
  id={undefined}
>
  <label
    className="FormField-label"
    htmlFor="737884497096766615"
  />
  <input
    className="FormField-form-control"
    id="737884497096766615"
  />
  <span
    className="FormField-note"
  >
    When I wore a younger mans clothes
  </span>
</div>
`;

exports[`<FormField> when props.help and props.isRequired are supplied renders help and required icon in the proper order 1`] = `
<div
  className="form-group"
  id={undefined}
>
  <label
    className="FormField-label"
    htmlFor="737884497096766615"
  >
    <span
      className="FormField-required"
      role="tooltip"
      title="required"
    >
      *
    </span>
    <Help
      id="help-737884497096766615"
      message="Please help me, please."
      overlayType="tooltip"
      rootClose={false}
      tooltipPlacement="top"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    />
  </label>
  <input
    className="FormField-form-control"
    id="737884497096766615"
  />
  <span
    className="FormField-note"
  />
</div>
`;

exports[`<FormField> when props.help is supplied and it is React.element renders a help icon 1`] = `
<div
  className="form-group"
  id={undefined}
>
  <label
    className="FormField-label"
    htmlFor="737884497096766615"
  >
    <Help
      id="help"
      message="Please help me, please."
      overlayType="tooltip"
      rootClose={false}
      tooltipPlacement="top"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    />
  </label>
  <input
    className="FormField-form-control"
    id="737884497096766615"
  />
  <span
    className="FormField-note"
  />
</div>
`;

exports[`<FormField> when props.help is supplied and it is simple text renders a help icon 1`] = `
<div
  className="form-group"
  id={undefined}
>
  <label
    className="FormField-label"
    htmlFor="737884497096766615"
  >
    <Help
      id="help-737884497096766615"
      message="Please help me, please."
      overlayType="tooltip"
      rootClose={false}
      tooltipPlacement="top"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    />
  </label>
  <input
    className="FormField-form-control"
    id="737884497096766615"
  />
  <span
    className="FormField-note"
  />
</div>
`;

exports[`<FormField> when props.isRequired is passed in and it is false does not render a required icon and title 1`] = `
<div
  className="form-group"
  id={undefined}
>
  <label
    className="FormField-label"
    htmlFor="737884497096766615"
  />
  <input
    className="FormField-form-control"
    id="737884497096766615"
  />
  <span
    className="FormField-note"
  />
</div>
`;

exports[`<FormField> when props.isRequired is passed in and it is true renders a required icon and title 1`] = `
<div
  className="form-group"
  id={undefined}
>
  <label
    className="FormField-label"
    htmlFor="737884497096766615"
  >
    <span
      className="FormField-required"
      role="tooltip"
      title="required"
    >
      *
    </span>
  </label>
  <input
    className="FormField-form-control"
    id="737884497096766615"
  />
  <span
    className="FormField-note"
  />
</div>
`;

exports[`<FormField> when props.label and labelAppendix are supplied renders a label with an appendix 1`] = `
<div
  className="form-group"
  id={undefined}
>
  <div
    className="FormField-labelWrapper"
  >
    <label
      className="FormField-label"
      htmlFor="737884497096766615"
    >
      label
    </label>
    <div
      className="FormField-labelAppendix"
    >
      <a>
        appendix
      </a>
    </div>
  </div>
  <input
    className="FormField-form-control"
    id="737884497096766615"
  />
  <span
    className="FormField-note"
  />
</div>
`;

exports[`<FormField> when props.label is supplied renders a labeled input 1`] = `
<div
  className="form-group"
  id={undefined}
>
  <label
    className="FormField-label"
    htmlFor="737884497096766615"
  >
    label
  </label>
  <input
    className="FormField-form-control"
    id="737884497096766615"
  />
  <span
    className="FormField-note"
  />
</div>
`;

exports[`<FormField> when the child component has a value for props.className will augment the child's props.className with the FormField-form-control class 1`] = `
<div
  className="form-group"
  id={undefined}
>
  <label
    className="FormField-label"
    htmlFor="737884497096766615"
  />
  <input
    className="FormField-form-control child-class"
    id="737884497096766615"
  />
  <span
    className="FormField-note"
  />
</div>
`;

exports[`<FormField> when the child component has a value for props.id will preserve the child component's id 1`] = `
<div
  className="form-group"
  id={undefined}
>
  <label
    className="FormField-label"
    htmlFor="child-id"
  />
  <input
    className="FormField-form-control"
    id="child-id"
  />
  <span
    className="FormField-note"
  />
</div>
`;
